Add favicon for the LaunchPad weblink and support attaching JIRA link in LP description.

hkhonming opened the pull request:
When you add the weblink manually in JIRA. It will automatically add the Launchpad favicon, which makes the link very visible to user. I am suggesting we do the same in the script

mclemenceau reviewed (commented):
@hkhonming Thanks for the contribution! I have no problem with the favicon, I think it is a good idea.
For the link details addition to Launchpad, I'm not so much in favor as LP is public and JIRA, in the context of Canonical is private so we would put links that no one can access.
Now at a minimum I would want lp-to-jira default behavior to not change and so the default option should be no link and you have to specify --lp-link to add the link instead of saying to don't want the link.
Hope this help!

hkhonming commented:
I originally want to separate the favicon commit but looks like I forgot to branch it out. So let's handle together.

Originally, the script will add a tag in LP by default. But in my opinion, the tag itself is not very useful since,
1. Usually the LP bug and JIRA is a 1-1 mapping and so the tag lost the filtering/grouping purpose in LP
2. User will still need to copy JIRA key and construct the URL with the prefix before they can redirect to the JIRA webpage.

I totally understand your concern on associating a public LP with a private JIRA. So in order to align the argument naming convention. I am suggesting to make both tag and description as OFF by default and just enable it with a positive arguments.

@mclemenceau what do you think with my update now?

mclemenceau reviewed (approved):
Approving that part of the PR as it is a nice enhancement indeed

mclemenceau reviewed (changes requested):
I meant to only apply the icon part of the changes but I guess this is all or nothing.
I recognized that the tag doesn't really make sense but some people might be using it.
Still I would love to not change the default behavior.

hkhonming commented:
So I have reverted the changes around --no-lp-tag so that it is just identical as before.

Since the argument I am adding is quite similar to --no-lp-tag but the default behaviour is different. To avoid confusion. I updated the argument name and description so that it should cause fewer confusion to user

mclemenceau reviewed (approved):
Looks good thank you!
